presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -418,10 +419,10 @@ public static Path createTemporaryPath(ConnectorSession session, HdfsContext con
         String temporaryPrefix = getTemporaryStagingDirectoryPath(session)
                 .replace("${USER}", context.getIdentity().getUser());
 
-        // use relative temporary directory on ViewFS
-        if (isViewFileSystem(context, hdfsEnvironment, targetPath)) {
-            temporaryPrefix = ".hive-staging";
-        }
+//        // use relative temporary directory on ViewFS
+//        if (isViewFileSystem(context, hdfsEnvironment, targetPath)) {
+//            temporaryPrefix = ".hive-staging";
+//        }
```

Comment:
why comment out this?

Reply:
Let me remove this piece of code, with "getTemporaryStagingDirectoryPath", this doesn't need anymore. Meanwhile, this .hive-staging even creates trouble, when you do create table from select, it will try to create .hive-staging dir, then mv it to the parent directory, which will make it exception since it will complain the directory already be there. (because when you create .hive-staging, the parent dir is already there).

So what TemporaryStagingDirectoryPath solves the exactly original problem, you can choose the common staging dir for the specific user by session parameter.